### ISSUE TITLE
Examples: amp-selector: No indication of focus

### ISSUE THREAD

**Issue by TetraLogicalHelpdesk** [ampproject/amphtml#31170]
## WCAG Level
Level AA

## Priority
High

## Type
Navigation

## Pages/screens/components affected
* [amp-base-carousel](https://amp.dev/documentation/components/amp-base-carousel/)
* amp-stream-gallery
* [amp-inline-gallery](https://amp.dev/documentation/components/amp-inline-gallery/)
* [amp-lightbox-gallery](https://amp.dev/documentation/components/amp-lightbox-gallery/)

## Description
Keyboard users rely on the browser's focus outline to keep track of where they are on web pages. The currently selected form input, button, or hyperlink is the one said to have the focus.

Most browsers use a faint dotted outline to indicate the focus to the user, though some browsers like Chrome or Safari use an enhanced focus outline, consisting of a colored border around the selected link. It is also possible for developers to provide custom focus indications within the site's cascading style sheets (CSS).

In the [`amp-selector` playground demo](https://playground.amp.dev/?url=https%3A%2F%2Fpreview.amp.dev%2Fdocumentation%2Fexamples%2Fcomponents%2Famp-selector&format=websites), none of the image-based options currently have a visible focus indication.

![Playground example of image-based amp-selector](https://user-images.githubusercontent.com/54933393/99181742-6749d880-2728-11eb-8f4b-b1bfd87c0764.png)

In the checkbox-like "Option" examples, the options do have a focus indicator - but not when they are checked. It's not possible to determine when a checked option (which by default has a border) also has focus.

![Playground example of image-based amp-selector](https://user-images.githubusercontent.com/54933393/99181744-6dd85000-2728-11eb-9e76-e4282484dfe6.png)

In the [`amp-selector` glitch demo](https://daisy-purring-seal.glitch.me/amp-selector.html), the options in "Single select (should be disabled)" section still receive focus, but lack a visible focus indication.

!['Single select (should be disabled)' set of listbox options](https://user-images.githubusercontent.com/54933393/99181750-7597f480-2728-11eb-8374-78ef745fc617.png)

In the "Multi select" and "Actions" sections, there is a visible focus indication, but not when focus goes to an already selected option - in those cases, only the border that is generally shown for selected options is visible.

!['Actions' set of listbox options - two options have a border, one of them has focus, but they look indistinguishable / no indication of focus](https://user-images.githubusercontent.com/54933393/99181754-7c266c00-2728-11eb-94b4-fc18e99391fe.png)

## User impact
Users who navigate using the keyboard will have difficulty operating the carousel/galleries, as it will not be clear to these users when the previous/next controls have focus. This problem will be further exacerbated in situations where the controls have been hidden (either due to an interaction using touch, or because the author has explicitly hidden them), as keyboard users will essentially see their focus "disappear" completely when they reach the invisible controls.

The default behaviour of permanently hiding the controls when the carousel/gallery has been operated via touch does not take into account users who may be using a touchscreen in conjunction with other input methods, such as a keyboard or mouse. They may be on a laptop/desktop with a touchscreen (e.g. Surface laptop), or use a "touch" device such as a smartphone or tablet, but have a paired Bluetooth mouse and/or keyboard. For users who occasionally use the touchscreen, but then switch to any of their other input mechanisms, the carousel/gallery will become unusable unless they, after the first touchscreen interaction, consistently only use touch to operate the carousels.

## Required solution
By default, the previous/next buttons must provide a visible focus indication. We recommend providing an explicit `:focus` style for these controls that is clear and visible - for instance, a stronger/thicker stroke/border.

Note that the `:focus` styles will also take effect after a mouse or touch user activated one of the options (which is likely why no focus styles have been implemented in the first place, at least on the image-based examples). To avoid this (as it may be visually confusing to mouse/touch users), consider using the new [`:focus-visible` pseudo-class](https://developer.mozilla.org/en-US/docs/Web/CSS/:focus-visible) in combination with the [`:focus-visible` polyfill](https://github.com/WICG/focus-visible).

In addition, make sure to also provide specific styles for options that are both focused *and* selected/checked.

## Test procedure(s)
Use these steps to confirm that the solution has been correctly applied to issues identified within the test sample, and to test the rest of the website for instances of the same issue:
1. Navigate through the page using a keyboard (`Tab`/`Shift`+`Tab`)
2. Verify that controls that receive focus have a visible focus indication
3. Verify that the keyboard focus does not "disappear" to any invisible controls
4. Interact with the options/controls to check/select them
5. Make sure that checked/selected options also have a visible focus indication when focused

## Related standards
* [WCAG 2.1 Success Criterion 2.4.7 Focus Visible (Level AA)](https://www.w3.org/TR/WCAG21/#focus-visible)

## More information
* [W3C - Failure of Success Criterion 2.4.7 due to styling element outlines and borders in a way that removes or renders non-visible the visual focus indicator](https://www.w3.org/WAI/WCAG21/Techniques/failures/F78.html)

## Test data
Test date: November 2020
